File: lab0104/app/src/main/java/com/example/lab0104/MainActivity2.java
package com.example.lab0104;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity2 extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        Intent intent = getIntent();
        TextView tv= findViewById(R.id.tv);
       // String smth = intent.getStringExtra("smth", smth);
        Bundle arguments = getIntent().getExtras();
        String text = arguments.get("smth").toString();
        tv.setText(text);
    }
}

File: lab0101/app/src/main/java/com/example/lab0101/MainActivity2.kt
package com.example.lab0101

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.widget.`AppCompatButton$InspectionCompanion`

class MainActivity2 : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        var btn: Button
        var txt : TextView

        btn=findViewById(R.id.mybtn2)
        txt=findViewById(R.id.mytxt2)

        btn.setOnClickListener {
            txt.setText("Goodbye, User")
        }
    }
}